### PR TITLE
[Feature] Migration from moment.js to date-fns

### PR DESCRIPTION
**Pull request by fnpen**
Moment.js:
- big and bloated

Done:
- Removed moment, added date-fns
- Replaced imports moment -> date-fns
- Removed .clone() / date-fns pure-functions
- Added babel-plugin-date-fns for better tree-shaking

**Results:**
rsuite.js  2.56 MiB -> rsuite.js  2.49 MiB **Δ -70KiB**
rsuite.min.js (573 KiB) -> rsuite.min.js (543 KiB) = **Δ-30KiB -5.2%**

Current state (https://bundlephobia.com/result?p=rsuite@3.5.3):
image

### CODE REVIEW

File: src/Calendar/MonthDropdown.js [hunk 1]
```diff
@@ -2,19 +2,20 @@
 
 import * as React from 'react';
 import classNames from 'classnames';
-import moment from 'moment';
 import List from 'react-virtualized/dist/commonjs/List';
 import AutoSizer from 'react-virtualized/dist/commonjs/AutoSizer';
 import { prefix, getUnhandledProps, defaultProps } from '../utils';
 import MonthDropdownItem from './MonthDropdownItem';
+import { getYear, setYear, setMonth, getMonth } from 'date-fns';
+import curry from '../utils/curry';
```
Comment (simonguo) [inline src/Calendar/MonthDropdown.js]: `lodash` already exists in lib, it is best to use `_.curry` directly.

Comment (fnpen) [reply src/Calendar/MonthDropdown.js]: I renamed curry to composeFunctions. lodash/curry - different helper.
But usefull for code understanding on piped execution.

**Comment by fnpen** [comment]
@simonguo conflict solved, ready to merge

**Comment by simonguo** [comment]
https://github.com/rsuite/rsuite/blob/97dfea05660cded3c4f63e4d7222d503de2e9483/src/utils/disabledTime.js#L22-L29

@fnpen 

`hours`,`minutes`,`seconds`  is not a function

**Comment by fnpen** [comment]
> [rsuite/src/utils/disabledTime.js](https://github.com/rsuite/rsuite/blob/97dfea05660cded3c4f63e4d7222d503de2e9483/src/utils/disabledTime.js#L22-L29)
> 
> Lines 22 to 29 in [97dfea0](/rsuite/rsuite/commit/97dfea05660cded3c4f63e4d7222d503de2e9483)
> 
>    return calendarProps[key](date.hours(), date); 
>  } 
>  if (/(Minutes)/.test(key)) { 
>    return calendarProps[key](date.minutes(), date); 
>  } 
>  if (/(Seconds)/.test(key)) { 
>    return calendarProps[key](date.seconds(), date); 
>  } 
> @fnpen
> 
> Missing `date-fns` migration to `disabledTime.js` file

Done, and also complete site migration and docs: https://github.com/rsuite/rsuite.github.io/pull/65

**Pull request merged**